### PR TITLE
Task8

### PR DESCRIPTION
**Pull request by meis0n**

### CODE REVIEW

File: UI/src/app/app.module.ts [hunk 1]
```diff
@@ -63,7 +67,20 @@ const appRoutes: Routes = [
     SharedModule,
     LoginModule,
   ],
-  providers: [],
+  providers: [{
+    provide: "BASE_API_URL",
+    useValue: 'http://localhost:3004'
```
Comment (AinuraAliieva) [inline UI/src/app/app.module.ts]: Consider keeping the value in `environment.ts`. To be able to set it depending on env.

File: UI/src/app/shared/services/authorization.service.ts [hunk 2]
```diff
@@ -1,24 +1,51 @@
...
+          login: data['login'],
+          firstName: data['name.first'],
+          lastName: data['name.last'],
+          email: data['login'],
+        })
+      )
+    );
+  }
+  private setToken(token: string): void {
+    return this.localStorage.set('AUTH', token);
```
Comment (AinuraAliieva) [inline UI/src/app/shared/services/authorization.service.ts]: Please create a separate variable for `AUTH` to follow DRY.

File: UI/src/app/courses/courses-page/courses-page.component.html [hunk 3]
```diff
@@ -3,15 +3,15 @@
     <div class="courses-page-actions">
       <div data-test-id="AddCourse" class="button button--secondary" (click)="addCourse()"><fa-icon class="icon" [icon] ="faPlus"></fa-icon>Add course</div>
     </div>
-    <ul *ngIf=courses.length>
-      <li *ngFor="let course of courses | filterByString:'title':filter | orderBy: 'creationDate'">
```
[the remark this answers is not in the document]
Comment (meis0n) [reply UI/src/app/courses/courses-page/courses-page.component.html]: I thought that was a temp feature while we doesn't any backend-filtering. Now backend returns filtered/ordered results. Should i return it?

[the remark this answers is not in the document]
Comment (AinuraAliieva) [reply UI/src/app/courses/courses-page/courses-page.component.html]: No, that's fine, let's keep as is.

File: UI/src/app/shared/services/authorization.service.ts (hunk 2 above)
Comment (meis0n) [reply UI/src/app/shared/services/authorization.service.ts]: done

File: UI/src/app/app.module.ts (hunk 1 above)
Comment (meis0n) [reply UI/src/app/app.module.ts]: done

File: UI/src/app/app.module.ts [hunk 4]
```diff
@@ -63,7 +67,20 @@ const appRoutes: Routes = [
     SharedModule,
     LoginModule,
   ],
-  providers: [],
+  providers: [{
+    provide: "BASE_API_URL",
```
[the remark this answers is not in the document]
Comment (meis0n) [reply UI/src/app/app.module.ts]: done

**Pull request merged**